src/Compiler/Checking/Expressions/CheckExpressions.fs

```diff
@@ -11220,6 +11188,30 @@ and TcNormalizedBinding declKind (cenv: cenv) env tpenv overallTy safeThisValOpt
 
         CheckedBindingInfo(inlineFlag, valAttribs, xmlDoc, tcPatPhase2, explicitTyparInfo, nameToPrelimValSchemeMap, rhsExprChecked, argAndRetAttribs, overallPatTy, mBinding, debugPoint, isCompGen, literalValue, isFixed), tpenv
 
+// Rotate [<return:...>] from binding to return value
+// Also patch the syntactic representation
+and TcNormalizeReturnAttribs cenv env attrTgt attrs valSynData rtyOpt =
+    let attribs = TcAttrs cenv env attrTgt attrTgt false attrs
+    let rotRetSynAttrs, rotRetAttribs, valAttribs =
+        // Do not rotate if some attrs fail to typecheck...
+        if List.length attribs <> attrs.Length then [], [], attribs
+        else attribs
+             |> List.zip attrs
+             |> List.partition(function | _, Attrib(_, _, _, _, _, Some ts, _) -> ts &&& AttributeTargets.ReturnValue <> enum 0 | _ -> false)
+             |> fun (r, v) -> (List.map fst r, List.map snd r, List.map snd v)
+    let retAttribs =
+        match rtyOpt with
+        | Some (SynBindingReturnInfo(attributes = Attributes retAttrs)) ->
+            rotRetAttribs @ TcAttrs cenv env attrTgt AttributeTargets.ReturnValue true retAttrs
```

Comment:
Doesn't this mean that the return attributes will be TcAttrs'd twice, leading to type checking and diagnostics duplicates?